### PR TITLE
Fix submesh logic and cache textures

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

The `lookforSubmesh` function was always returning a valid object, when it should return 'false` if the submesh was not found. I've modified the function to return a pair `[boolean, obj]`, where the boolean is true when the submesh was found and false otherwise.

I've also update the `loadTexture` function to cache textures.

### CODE REVIEW

File: src/Scene.ts [hunk 1]
```diff
@@ -3485,45 +3488,54 @@ export class Scene {
...
+          // Create the image element
+          let imageElem: HTMLImageElement = <HTMLImageElement>(
+            document.createElementNS('http://www.w3.org/1999/xhtml', 'img'));
 
-         var binary = '';
-         var len = image.byteLength;
-         for (var i = 0; i < len; i++) {
-           binary += String.fromCharCode(image[i]);
-         }
+          var isJPEG = map.search( /\.jpe?g($|\?)/i ) > 0 || map.search( /^data\:image\/jpeg/ ) === 0;
```
Comment (german-e-mas) [inline src/Scene.ts:3507]: Avoid using `var`s, use `let` or `const`.

**Review by german-e-mas — APPROVED** [review]
LGTM! I left some comments for things to take into account in the future, no need to address them right now.

File: src/Scene.ts [hunk 2]
```diff
@@ -3485,45 +3488,54 @@ export class Scene {
     printGraph(this.scene);
   }
 
- public loadTexture(url: string, onLoad?: any, onProgress?:any): THREE.Texture {
-   let scope = this;
+  public loadTexture(url: string, onLoad?: any, onProgress?:any): THREE.Texture {
+    let scope = this;
```
[the remark this answers is not in the document]
Comment (nkoenig) [reply src/Scene.ts:3492]: Ahh. I didn't know arrow functions were fundamentally different. Good to know. I'll make that change here.

File: src/Scene.ts (hunk 1 above)
Comment (nkoenig) [reply src/Scene.ts]: [484d2a6](https://github.com/gazebo-web/gzweb/pull/14/commits/484d2a63510457fede28f6772d0a4c65d0b4c582)

**Pull request merged**